Implemented LegacySerializer for the TileSet type.

swordmaster2k opened the pull request:
Implemented LegacySerializer for the TileSet type.

Removed support for tiles in AnimationFrames will encourage users to use image files instead.

Signed-off-by: Joshua Michael Daly <b00056835@student.itb.ie>

swordmaster2k committed:
Implemented LegacySerializer for the TileSet type.

Removed support for tiles in AnimationFrames will encourage users to use image files instead.


Signed-off-by: Joshua Michael Daly <b00056835@student.itb.ie>

goblinJoel commented on src/main/java/net/rpgtoolkit/common/assets/TileSet.java:
```diff
@@ -189,170 +93,82 @@ public int getTileHeight() {
-      System.out.println(e.toString());
-      return false;
-    }
+  public boolean isHasChanged() {
```
Wut?

swordmaster2k replied:
Lol, auto-generated getter will refactor :)

swordmaster2k replied:
It was actually unused so I just removed it

goblinJoel commented on src/main/java/net/rpgtoolkit/common/utilities/TileSetCache.java:
```diff
@@ -20,7 +27,7 @@
 
   // Singleton.
-  private static final TileSetCache instance = new TileSetCache();
+  private static final TileSetCache INSTANCE = new TileSetCache();
```
Since this is already static and already initializing regardless of whether it gets used, can we just make this all static, or is there a benefit to having this be a singleton with an actual instance?

swordmaster2k replied:
You mean like change all its members and methods to static? 

I dunno when I wrote this class I just chose the Singleton pattern by default. Making it static won't have much impact really, right?

goblinJoel replied:
Yes. It'll make it easier to use (just TileSetCache.foo() instead of TileSetCache.getInstance().foo()), but otherwise I'm not aware of anything. Singleton was new to me when I started on this project, but my understanding from reading a bit is that it's not super necessary here, given the way Java does static.

Personally, I think it's easier to think of it as "The tile set cache" rather than, "The tile set cache's instance (which we could make more of, but we shouldn't, so that ability is private)".

swordmaster2k replied:
Converted it to a Static class a re-tested works the same as before. I think there is a slight performance boost when using static methods from what I can remember.

goblinJoel commented on src/main/java/net/rpgtoolkit/common/assets/AnimationFrame.java:
```diff
@@ -56,16 +56,6 @@ public AnimationFrame(String frameName, long transparentColour, String frameSoun
       }
-    } else {
-      // Loads the tileset and fetches the correct tile
-      frameName = frameName.toLowerCase();
```
Since tiles are no longer for animation frames, what things are they still good for? Boards?

swordmaster2k replied:
They are used in Animations, they are made up of AnimationFrames, easier just to say they are created from png, jpgs, gifs. 

Having the option to use individual tiles from tilesets over complicates things.

goblinJoel replied:
Animation frames are made of images, tiles can be made of animation frames, and animations are made of animation frames. Animation frames used to be able to be made of tiles, too, but we removed that. Did I get that right?

goblinJoel reviewed (changes requested):
So, this moves the tileset serialization to its own legacy serializer and switches the sort of stream/buffer thing it uses to do that. It also removes the ability to use tiles in animation frames. See comments for questions.

Only actual change I insist on is renaming "isHasChanged()" to something that isn't "isHasChanged()".

swordmaster2k committed:
#12

Converted TileSetCache from Singleton to Static class.

Removed unused hasChanged attribute from TileSet type.

Signed-off-by: Joshua Michael Daly <b00056835@student.itb.ie>

goblinJoel commented on src/main/java/net/rpgtoolkit/common/assets/TileSet.java:
```diff
@@ -25,8 +25,6 @@
-
   private int tilesetVersion;
   private int numberOfTiles;
   private int tilesetType;
```
This may be out of scope for this PR, but should we just use LinkedLists.size() instead of tracking numberOfTiles separately?

swordmaster2k replied:
Will be covered in FileFormat restructuring so I'll leave it for this PR.

goblinJoel commented on src/main/java/net/rpgtoolkit/common/utilities/TileSetCache.java:
```diff
@@ -26,32 +26,17 @@
-  public static TileSetCache getInstance() {
-    return INSTANCE;
-  }
+  private static final HashMap<String, TileSet> TILE_SETS = new HashMap<>();;
```
The ; is doubled here.

swordmaster2k committed:
Removed a double ;.

#12

Signed-off-by: Joshua Michael Daly <b00056835@student.itb.ie>

swordmaster2k commented:
All issues resolved, merging this now.

goblinJoel commented:
Good work.

That last commit wins the smallest commit award!
